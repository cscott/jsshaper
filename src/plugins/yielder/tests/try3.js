var log = (typeof console !== "undefined") && console.log || print;
// adjust path to get generator.js
require && require.paths && typeof __dirname !== "undefined" && require.paths.unshift(__dirname+"/..");
StopIteration = require('stopiteration.js');

function foo() {
    log("1");
    try {
        log('2');
        throw 3;
    } catch (e) {
        log(e);
        throw 4;
    } finally {
        log('finally');
        throw 7;
    }
}

function bar(b, c) {
    log("1");
    try {
        log('2');
        if (b) throw 3;
    } catch (e) {
        log(e);
        throw 4;
    } finally {
        log('finally');
        if (c) throw 7;
    }
    log('after');
    yield 8;
}

try {
    log(foo());
    log('x');
} catch (e) {
    log(e);
}

try {
    log(bar(true, true).next());
    log('x');
} catch (e) {
    log(e);
}

try {
    log(bar(false,false).next());
    log('x');
} catch (e) {
    log(e);
}
