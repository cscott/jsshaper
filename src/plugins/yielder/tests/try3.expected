1
2
3
finally
7
1
2
3
finally
7
1
2
finally
after
8
x
